azure-devops/azext_devops/dev/team/project.py

```diff
@@ -94,40 +94,36 @@ def create_project(name, devops_organization=None, process=None, source_control=
         raise CLIError(ex)
 
 
-def delete_project(project_id=None, devops_organization=None, detect=None):
+def delete_project(id, devops_organization=None, detect=None):
     """Delete team project.
-    :param project_id: The id (UUID) of the project to delete.
-    :type project_id: str
+    :param id: The id (UUID) of the project to delete.
+    :type id: str
     :param devops_organization: Azure Devops organization URL. Example: https://dev.azure.com/MyOrganizationName/
     :type devops_organization: str
     :param detect: When 'On' unsupplied arg values will be detected from the current working
                    directory's repo.
     :type detect: str
     """
     try:
-        if project_id is None:
-            raise CLIError('--id argument needs to be specified.')
         devops_organization = resolve_instance(detect=detect, devops_organization=devops_organization)
         core_client = get_core_client(devops_organization)
-        operation_reference = core_client.queue_delete_project(project_id=project_id)
+        operation_reference = core_client.queue_delete_project(project_id=id)
         operation = wait_for_long_running_operation(devops_organization, operation_reference.id, 1)
         status = operation.status.lower()
         if status == 'failed':
             raise CLIError('Project deletion failed.')
         elif status == 'cancelled':
             raise CLIError('Project deletion was cancelled.')
-        print('Deleted project {}'.format(project_id))
+        print('Deleted project {}'.format(id))
         return operation
     except VstsServiceError as ex:
         raise CLIError(ex)
 
 
-def show_project(project_id=None, name=None, devops_organization=None, detect=None, open_browser=False):
+def show_project(id=None, devops_organization=None, detect=None, open_browser=False):
```

Comment:
Id is compulsory now. Also now there is no check in the code to handle if it is none.

Reply:
nice catch made it madatory